src/main/java/mate.academy/Main.java

```diff
@@ -10,7 +10,14 @@ public class Main {
     public static void main(String[] args) {
         // Please test your Injector here. Feel free to push this class as a part of your solution
         Injector injector = Injector.getInjector();
-        ProductService productService = null;
+        ProductService productService = (ProductService) injector.getInstance(ProductService.class);
+
+        // Checking the singleton pattern (on AI mentor request)
+        Injector injectorSecond = Injector.getInjector();
+        if (!injector.equals(injectorSecond)) {
+            throw new RuntimeException("The singleton pattern doesn't work correctly");
+        }
```

Comment:
It's not necessary to test the singleton pattern functionality within the main method of the application. This should be covered by dedicated unit tests.